src/wazuh_modules/wm_gcp.c

```diff
@@ -477,7 +519,36 @@ void wm_gcp_bucket_run(const wm_gcp_bucket_base *data, wm_gcp_bucket *exec_bucke
 
     os_free(output);
 }
+#ifdef WIN32
+DWORD WINAPI wm_gcp_pubsub_destroy(void *gcp_config) {
+    wm_gcp_pubsub *data = (wm_gcp_pubsub *)gcp_config;
+
+    if (data->project_id) os_free(data->project_id);
+    if (data->subscription_name) os_free(data->subscription_name);
+    if (data->credentials_file) os_free(data->credentials_file);
+    os_free(data);
+    return 0;
+}
 
+DWORD WINAPI wm_gcp_bucket_destroy(void *gcp_config) {
+    wm_gcp_bucket *cur_bucket;
+    wm_gcp_bucket_base * data = (wm_gcp_bucket_base *)gcp_config;
+    wm_gcp_bucket *next_bucket = data->buckets;
+
+    while(next_bucket){
+        cur_bucket = next_bucket;
+        next_bucket = next_bucket->next;
+        if (cur_bucket->bucket) os_free(cur_bucket->bucket);
+        if (cur_bucket->type) os_free(cur_bucket->type);
+        if (cur_bucket->credentials_file) os_free(cur_bucket->credentials_file);
+        if (cur_bucket->prefix) os_free(cur_bucket->prefix);
+        if (cur_bucket->only_logs_after) os_free(cur_bucket->only_logs_after);
+        os_free(cur_bucket);
+    }
+    os_free(data);
+    return 0;
```

Comment:
this could be summarized with a #ifdef WIN32 on the return to avoid the entire duplication of the function.